src/clients/c/tb_client/echo_client.zig

```diff
@@ -62,32 +68,70 @@ pub fn EchoClient(comptime StateMachine_: type, comptime MessageBus: type) type
             self.reply();
         }
 
-        pub fn request(
+        pub fn batch_get(
             self: *Self,
-            user_data: u128,
-            callback: Self.Request.Callback,
             operation: Self.StateMachine.Operation,
-            message: *Message,
-            message_body_size: usize,
-        ) void {
-            const message_request = message.build(.request);
+            event_count: usize,
+        ) Self.BatchError!Batch {
+            const body_size = switch (operation) {
+                inline else => |op| @sizeOf(Self.StateMachine.Event(op)) * event_count,
```

Comment:
```suggestion
                inline else => |operation| @sizeOf(Self.StateMachine.Event(operation)) * event_count,
```

"op" typically refers to an op number, not the state machine operation type.